### PR TITLE
SafeQuit() refactor

### PR DESCRIPTION
**Pull request by Crotalus**
Don't call both ExitGame() and ExitApplication(). The former is called for
non-online stuff.

Also, the function was in "c-boundary" so WaitSeconds() always failed,
thus risking SessionEndGame() not to complete.

Maybe this helps with desynced / corrupted replays, needs to be tested
ofc.

### CODE REVIEW

**Comment by Sheeo** [comment]
:+1:

Could this target release/3641 instead though?

**Comment by Crotalus** [comment]
Hmm, is it possible to retarget or need new PR?

File: lua/ui/dialogs/eschandler.lua [hunk 1]
```diff
@@ -12,19 +12,25 @@ local Utils = import('/lua/system/utils.lua')
...
+        return
+    end
+
+    if SessionIsActive() then
+        ForkThread(function ()
+            ConExecute('ren_oblivion true')
+            ConExecute('ren_ui false')
+            SessionEndGame()
+            WaitSeconds(1)
+            ExitApplication()
```
Comment (Sheeo) [inline lua/ui/dialogs/eschandler.lua]: This may indeed be the way to go.

**Comment by Sheeo** [comment]
Yes, we would need a new PR.

But let's just merge it into release/3641 and close the PR afterwards. I give my :+1:

**Pull request closed without merging**